### PR TITLE
pkg/deploy: Add the empty openshift.io/node-selector annotation to the metering-operator namespace.

### PR DESCRIPTION
**Pull request by timflannagan1**
In the case where a cluster administrator has configured the `defaultNodeSelector` field for the cluster's `Scheduler` object, which is responsible for determining the placement of Pods in a cluster, there may be conflicts when a user also specifies a `nodeSelector` configuration for the metering-operator's operands in the `MeteringConfig` object. Instead, if we always set the `'openshift.io/node-selector'` annotation to be empty, meaning that we don't care where the Pods in this namespace get scheduled, the `defaultNodeSelector` won't be injected by the Scheduler and the user-provided `nodeSelector` configuration will work as intended.

Note: these changes are only geared towards manually deploying Metering manually (e.g. deploy everything ourselves) and deploying Metering through OLM manually. In order to install Metering through OperatorHub, you would still need to create/label/annotate the namespace yourself as a pre-requisite step to configuring Metering.

### CODE REVIEW

File: pkg/deploy/install.go [hunk 1]
```diff
@@ -29,6 +29,10 @@ func (deploy *Deployer) installNamespace() error {
 		if deploy.config.Platform == "openshift" {
 			labels["openshift.io/cluster-monitoring"] = "true"
 			deploy.logger.Infof("Labeling the %s namespace with 'openshift.io/cluster-monitoring=true'", deploy.config.Namespace)
+			namespaceObjectMeta.Annotations = map[string]string{
```
Comment (bentito) [inline pkg/deploy/install.go:32]: I think a comment along the lines of the commit message would be helpful to explain what's going on.

Comment (timflannagan1) [reply pkg/deploy/install.go:51]: Updated in the last force-push. I tried to add some context to both of the namespace metadata modifications we make.

Comment (bentito) [reply pkg/deploy/install.go:51]: image

**Comment by timflannagan1** [comment]
Let me file a BZ for that upgrade e2e failure
/hold

**Comment by timflannagan1** [comment]
Filed a BZ for the upgrade failure flake. Going to override that test and manually add the valid-bz label as this Go library is only used internally + testing purposes.
/hold cancel
/override ci/prow/metering-e2e-aws

**Pull request merged**